internal/cmd/update/command.go

```diff
@@ -74,6 +80,21 @@ func NewClient(cliName string, disableUpdateCheck bool) update.Client {
 }
 
 func (c *command) update(cmd *cobra.Command, _ []string) error {
+	if c.Config.DisableUpdates {
+		message := "updates are disabled for this binary"
+		if isHomebrew() {
+			return errors.NewErrorWithSuggestions(
+				message,
+				fmt.Sprintf("If installed with Homebrew, run `brew upgrade %s`.", homebrewFormula),
+			)
+		}
+
+		return errors.NewErrorWithSuggestions(
```

Comment:
The only thing I'm wondering here is if we want to always default to the assumption that they're using the CP installed CLI, or if we want to try and detect this too. Mainly just in case other checks before this report false negatives (e.g. `EvalSymLinks` can fail in some cases, though I'm not sure if they're applicable to us).

There's already logic in `connect plugin install` that examines the CLI's path from `os.Executable()` to determine if it's part of a package manager or archive CP installation.

Reply:
> EvalSymLinks can fail in some cases

Good point, we should show a generic error in this case instead of a CP-specific error. I'll let CP detection be a separate PR if we want to go down that route.